pyqtorch/apply.py

```diff
@@ -57,3 +59,32 @@ def apply_operator(
         map(lambda e: "".join(list(e)), [operator_dims, in_state_dims, out_state_dims])
     )
     return einsum(f"{operator_dims},{in_state_dims}->{out_state_dims}", operator, state)
+
+
+def operator_product(op1: Tensor, op2: Tensor, target: int) -> Tensor:
+    """
+    Compute the product of two operators.
+
+    Args:
+        op1 (Tensor): The first operator.
+        op2 (Tensor): The second operator.
+        target (int): The target qubit index.
+
+    Returns:
+        Tensor: The product of the two operators.
+    """
+
+    n_qubits_1 = int(log2(op1.size(1)))
+    n_qubits_2 = int(log2(op2.size(1)))
+    batch_size_1 = op1.size(-1)
+    batch_size_2 = op2.size(-1)
+    if n_qubits_1 > n_qubits_2:
+        op2 = promote_operator(op2, target, n_qubits_1)
+    elif n_qubits_1 < n_qubits_2:
```

Comment:
Would there be a  case where both operators are of the same size in the else?

Reply:
This code part is intended to ensure that both tensors are the same `size` before `torch.bmm()` to prevent errors. If both tensors already have the same `size` it is not necessary to handle them and you can directly run `torch.bmm()`. 
This case is particularly possible when studying a 1 qubit system because the doors are mostly defined on a 1 qubit.